MovieViewer/FirebaseUserDataService.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Firebase.swift
+//  MovieViewer
+//
+//  Created by hvmark on 7/11/16.
+//  Copyright © 2016 Dang Quoc Huy. All rights reserved.
+//
+
+import Firebase
+
+//class FirebaseUserDataService: UserDataServiceProtocol {
+class FirebaseUserDataService {
+
+    static func userFavoriteMovie(username: String, movieID: Int) {
```

Comment:
I see that you do immediately call `didFavorite` from the view controller, but that has limitations, you can't ever recover if an error occurs.